packages/vike-node/package.json

```diff
@@ -48,39 +86,26 @@
     "elysia": "^1.1.6",
     "fastify": "^4.28.1",
     "h3": "^1.12.0",
-    "hono": "^4.5.5",
+    "hono": "^4.6.3",
+    "tsup": "^8.3.0",
     "typescript": "^5.5.4",
-    "vike": "^0.4.193",
-    "vite": "^5.4.0"
-  },
-  "typesVersions": {
-    "*": {
-      "connect": [
-        "./dist/connect.d.ts"
-      ],
-      "fastify": [
-        "./dist/fastify.d.ts"
-      ],
-      "h3": [
-        "./dist/h3.d.ts"
-      ],
-      "hono": [
-        "./dist/hono.d.ts"
-      ],
-      "elysia": [
-        "./dist/elysia.d.ts"
-      ],
-      "plugin": [
-        "./dist/plugin/index.d.ts"
-      ],
-      ".": [
-        "./dist/index.d.ts"
-      ]
-    }
+    "universal-middleware": "^0.3.8",
+    "vike": "^0.4.198",
+    "vite": "^5.4.8"
   },
   "files": [
     "dist/"
   ],
   "repository": "github:vikejs/vike-node",
-  "license": "MIT"
-}
+  "license": "MIT",
+  "optionalDependencies": {
```

Comment:
Shouldn't these be peer dependencies of `vike-node`? Alternatively, if they are all relatively small, then I'd say let's add them to `dependency` so that the user doesn't have to install these himself.

Reply:
Those are generated by `universal-middleware`. I chose `optionalDependencies` because of its [documented behaviour](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#optionaldependencies):
> If a dependency can be used, but you would like npm to proceed if it cannot be found or fails to install, then you may put it in the optionalDependencies object. This is a map of package name to version or url, just like the dependencies object. The difference is that build failures do not cause installation to fail

[`pnpm`](https://github.com/pnpm/pnpm/issues/5928) and others seems to follow the exact same rule.

I think this is slightly more correct for what `universal-middleware` is doing, but I can be wrong.

Reply:
> Or we make this repo a monorepo and replace:
> 
>     * `vike-node/express` => `vike-express`
> 
>     * `vike-node/hono` => `vike-hono`
> 
>     * ...
> 
> 
> I like this most so far. Seems to be the cleanest and the most future-proof. I'm happy to elaborate if you want me to.

I do not have any preference about this (perhaps just that `vike-node` is too restrictive now as a name, because it handles more than node now).

> Why would the installation fail?

I can fail for many reason (network, or willingly blocking some packages from installing).

> Assuming it doesn't fail then I guess it's equivalent to `dependencies`? I'd rather keep it simple in that case.

Yes. This would require a modification of `universal-middleware` too (could be an option).

Reply:
> I do not have any preference about this (perhaps just that `vike-node` is too restrictive now as a name, because it handles more than node now).

Actually, let's keep `vike-node/{express,hono,...}` for now: we'll soon see whether/how relevant breaking it down in `vike-{express,hono,...}` is, and maybe we'll realizie that something else is better such as renaming `vike-node` to `vike-server` or having `vike-node` be built into Vike core. Let' see.

> I can fail for many reason (network, or willingly blocking some packages from installing).

Seems like a fringe situation? The same could be said for any other dependency as well? (I understand that from all adapters only one or two are needed by the user.)

AFAICT using `optionalDependencies` is adding more confusion than adding real benefits.

> Yes. This would require a modification of `universal-middleware` too (could be an option).

I wonder whether it could make sense to remove the usage of `optionalDependencies` throughout `universal-middleware`.

Reply:
`universal-middleware` could pre-bundle everything. As all bundles are quite slim (and even if it were not the case, the bloat would only be in dev), I should perhaps do that. No more need for any reference to `universal-middleware` sub packages that way.

Reply:
If these adapters are small then I guess so :+1:

Reply:
Nevermind, adding them as `dependencies` will achieve the same effect and doesn't require custom code.

Reply:
EDIT: I could do both those thing in fact without much work. Let me try something

Reply:
> EDIT: I could do both those thing in fact without much work. Let me try something

Yea I was thinking that as well.

If we go with `vike-{express,hono,...}` then cherry-picking adapters would be slightly better.

But for `vike-node` having an adapter bundle could be convenient, maybe. Although adding a new adapter to `vike-node` requires changes to its source code anyways. So, at the end of the day, I ain't sure what added value an adapter bundle would be :eyes:

Reply:
> One thing to consider: users may be inclined to think that universal-middleware is "dirty/heavy" because it includes all adapters.

I think that most users do not bother about those things, and for those who do, we can have a NOTE in the documentation stating that it's not a concern.

Reply:
:+1: As you wish.

(In case you're curious. I think that, at the end of the day, what the typical/average Next.js user wants is a working tool. But I also think that Vike users are more inclined to care about using conceptually well-designed tools, and also that selling Vike as "lightweight" is an argument for a non-negligible amount of users. E.g. Evan You cares a lot about making Vite as lightweight as possible.)